internal/controller/gitrepository_controller.go

```diff
@@ -957,22 +957,82 @@ func (r *GitRepositoryReconciler) verifyCommitSignature(ctx context.Context, obj
 	for _, v := range secret.Data {
 		keyRings = append(keyRings, string(v))
 	}
-	// Verify commit with GPG data from secret
-	entity, err := commit.Verify(keyRings...)
-	if err != nil {
-		e := serror.NewGeneric(
-			fmt.Errorf("signature verification of commit '%s' failed: %w", commit.Hash.String(), err),
-			"InvalidCommitSignature",
+
+	// these flags help us later figure out which objects we need to verify
+	var verifyHead bool
+	var verifyTag bool
+	switch obj.Spec.Verification.GetMode() {
+	case sourcev1.ModeHEAD:
+		verifyHead = true
+	case sourcev1.ModeTag:
+		verifyTag = true
+	case sourcev1.ModeTagAndHEAD:
+		verifyHead = true
+		verifyTag = true
+	}
```

Comment:
Instead of setting booleans using a `case` switch, why not just check the mode directly in the `if` statements for `verifyHead` and `verifyTag`?

Reply:
mainly to avoid long if statements, i can check the mode directly if you feel like that's better, i initially had that but i refactored it later to this because i thought this is more readable

Reply:
I think think this could partly be factored out then by introducing a `HasMode` in the API package?

Reply:
wdyt?

```go
func (m GitVerificationMode) VerifiesObject(object string) bool {
	if object == "commit" && (m == ModeHEAD || m == ModeTagAndHEAD) {
		return true
	}
	if object == "tag" && (m == ModeTag || m == ModeTagAndHEAD) {
		return true
	}
	return false
}
```

Reply:
`VerifyObjectType`?

Reply:
cc: @darkowlzz, may be handy to have an extra opinion here.

Reply:
Thinking more about this, the `object` parameter has no actual meaning within the API itself. Instead, I think you could get pretty far simplifying the `if`-logic by introducing a `GitVerificationMode#Equals` and `GitRepositoryRef#IsTag`.